### ISSUE TITLE
RetryRequestExecutor does not allow maxRetries to be 0

### ISSUE THREAD

**Issue by herospYL** [okta/okta-sdk-java#330]
According to the doc here: https://github.com/okta/okta-sdk-java/#connection-retry--rate-limiting It should support setting maxRetries to 0.

However, in class RetryRequestExecutor, the constructor does not allow setting this value to 0.

Tested with both Java System properties and grammatically (with Clients.builder().setRetryMaxAttempts(0))

**Comment by bdemers** [comment]
Hi @herospYL !

I’m not sure I fully understand your question. Are you getting an error?

**Comment by herospYL** [comment]
I’m expecting the value is able to set to 0. The current behavior is the constructor ignores 0 value, and uses the default 4 as the value. See Line 55 in https://github.com/okta/okta-commons-java/blob/03318517f44d0f8eabef1522310cb334587307c1/http/http-api/src/main/java/com/okta/commons/http/RetryRequestExecutor.java#L123

**Comment by bdemers** [comment]
I see it now! Thanks @herospYL We will get that fixed!

**Comment by herospYL** [comment]
Hi,

I'd like to know when do your team plan to fix this? Since I believe it's just changing from > to >=, I'd like to know if it's okay for me to directly contribute to the repository and create a PR

Thanks

**Comment by bdemers** [comment]
PRs are always welcome 😁

**Comment by peeyushsurolia** [comment]
Any update on this issue? @herospYL Did you get chance to create a PR for this.

**Comment by bdemers** [comment]
@peeyushsurolia they did :)
https://github.com/okta/okta-commons-java/pull/30

**Comment by peeyushsurolia** [comment]
Thanks @bdemers  for update. The one scenario I would like to discuss here related to that may be we can do some testing around that.

```
Clients.builder()
                    .setOrgUrl(checkNotNull(baseUrl))
                    .setRetryMaxAttempts(0)
                    .setConnectionTimeout(connectionTimeout)
                    .setRetryMaxElapsed(1)
                    .setClientCredentials(new TokenClientCredentials(checkNotNull(apiToken)))
```


For above code I ran few scenarios and got different behavior

|| **ConnectionTimeOut**  || **Actual timeout**   || **Expected Timeout** ||
| 1 sec                           | 4 sec                    | 1 sec                       |
| 2 sec                           | 6 sec                    | 2 sec                       |
| 3 sec                           | 9 sec                    | 3 sec                       |
| 4 sec                           | 12 sec                  | 4 sec                       |
| 5 sec                           | 15 sec                  | 5 sec                       |

I believe it is a exponential back off time which actually happening with every retry based on connectionTimeout.

Could you please confirm if this will get resolved after this fix?

Thanks.

**Comment by bdemers** [comment]
@peeyushsurolia can do!
cc @arvindkrishnakumar-okta

**Comment by arvindkrishnakumar-okta** [comment]
This will be fixed with v2.0.0 upcoming release.

**Comment by arvindkrishnakumar-okta** [comment]
Released v2.0.0 yesterday. Please try it out. Closing this issue now, feel free to reopen if follow-up is needed.
